flang/lib/Lower/ConvertCall.cpp

```diff
@@ -202,6 +205,8 @@ fir::ExtendedValue Fortran::lower::genCallOpAndResult(
     llvm::SmallVector<mlir::Value> lengths;
     if (!caller.callerAllocateResult())
       return {};
+    if (isProcedurePointer)
```

Comment:
This is to respond the above comment at line 162. 
This change is actually no longer needed as I don't `setSaveResult` for procedure pointer function result. Therefore, call to `caller.callerAllocateResult` is returned `false` for procedure pointer function result.